### PR TITLE
EFI-154 - add more descriptive error message on OAuth error

### PR DESCRIPTION
**Pull request by martinb89**

### CODE REVIEW

File: pEpForiOS/Util/OAuth2/OAuth2AuthorizationProtocol.swift [hunk 1]
```diff
@@ -7,17 +7,67 @@
 //
 
 import Foundation
-
```
[the remark this answers is not in the document]
Comment (martinb89) [reply pEpForiOS/Util/OAuth2/OAuth2AuthorizationProtocol.swift:10]: Ok, just for our information, we have 27 cases where we don't have an empty line in between and 25 where we do. 
Will revert the deletion of the empty line.

File: pEpForiOS/Util/OAuth2/OAuth2AuthorizationProtocol.swift [hunk 2]
```diff
@@ -7,17 +7,67 @@
...
 import MessageModel
 
-/**
- The error that gets delegated when there was no error during authorization, but
- neither a valid access token.
- */
-enum OAuth2AuthorizationError: Error {
+// We might want to distinguish error types.
+enum OAuthErrorType: String {
```
[the remark this answers is not in the document]
Comment (martinb89) [reply pEpForiOS/Util/OAuth2/OAuth2AuthorizationProtocol.swift:13]: Removed.

File: pEpForiOS/Util/OAuth2/OAuth2AuthorizationProtocol.swift [hunk 3]
```diff
@@ -7,17 +7,67 @@
...
+    init(error: Error?) {
+        self.domain = (error as? NSError)?.domain
+        self.errorDescription = (error as? NSError)?.userInfo["NSUnderlyingError"] as? String
+        self.type = .inconsistentAuthorizationResult
+        let code = (error as? NSError)?.code ?? -1
+        self.errorMessage = OAuth2AuthorizationError.getErrorMessage(code: code)
+    }
+
+    static private func getErrorMessage(code: Int) -> String {
+        var message = ""
```
[the remark this answers is not in the document]
Comment (martinb89) [reply pEpForiOS/Util/OAuth2/OAuth2AuthorizationProtocol.swift:34]: Done

File: pEpForiOS/Util/OAuth2/OAuth2AuthorizationProtocol.swift [hunk 4]
```diff
@@ -7,17 +7,67 @@
...
+        switch code {
+        case OIDErrorCode.invalidDiscoveryDocument.rawValue:
```
[the remark this answers is not in the document]
Comment (martinb89) [reply pEpForiOS/Util/OAuth2/OAuth2AuthorizationProtocol.swift:36]: I took it from the documentation of AppAuth in the file OIDError.h.
I will leave something shorter.

**Review by planckdz — APPROVED** [review]

**Comment by martinb89** [comment]
Merged

**Pull request closed without merging**